conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/AsyncGenerator.java

```diff
@@ -132,8 +132,8 @@ public MethodSpec generate(ServiceDefinition def) {
     }
 
     private Optional<FieldSpec> deserializer(EndpointName endpointName, Optional<Type> type) {
-        // TODO(forozco): handle optional binary ??
-        if (type.map(t -> t.accept(TypeVisitor.IS_BINARY)).orElse(false)) {
+        if (type.map(t -> t.accept(TypeVisitor.IS_BINARY) || isOptionalBinary(t))
```

Comment:
can we factor this lambda out into a separate function?

Reply:
Extracted this with support for dealiasing.